### PR TITLE
Adds support for Elasticsearch 5

### PR DESCRIPTION
**Pull request by rikatz**
This PR adds support to Elasticsearch 5, according to issue #1410 

It has been tested on Elasticsearch 2 and on Elasticsearch 5 and works as expected.

There are three major changes:

1) Changes the referenced library from elastic.v3 to elastic.v5
2) Adds the context on validation functions (Do(context.Background()), as this is requested by the new library.
3) Changes the Godeps, removing old vendor and adding the new one.

### CODE REVIEW

**Comment by rikatz** [comment]
I signed it

**Comment by rikatz** [comment]
Now I really signed it :P

**Comment by DirectXMan12** [comment]
@AlmogBaku / @huangyuqi / @andyxning please review

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"ImportPath": "k8s.io/heapster",
-	"GoVersion": "go1.6",
+	"GoVersion": "go1.7",
```
Comment (DirectXMan12) [inline Godeps/Godeps.json]: This doesn't actually change the compiler, IIRC -- it's basically just a record of which version of go was used to run Godep.  Please set it back.

**Comment by rikatz** [comment]
@DirectXMan12 Thanks for the answer! Should I 'git revert' this, or open another PR?

Thanks!

**Comment by AlmogBaku** [comment]
@rikatz please tag me there

**Comment by rikatz** [comment]
@AlmogBaku sure, I'm just testing stuff here and will open the new PR ASAP

**Pull request closed without merging**